### PR TITLE
Upgrade to DSL-based SpeziBluetooth

### PR DESCRIPTION
**Pull request by Supereg**
# Upgrade to DSL-based SpeziBluetooth

## :recycle: Current situation & Problem
This PR upgrades to the latest Spezi 1.0 releases. Further, this upgrades to the upcoming SpeziBluetooth release that introduces a new and simplified DSL API for declaring Bluetooth devices.

## :gear: Release Notes 
* Upgrade to latest Spezi 1.0 releases
* Upgrade to upcoming SpeziBluetooth releases.
* Refactor the device model supporting three different devices types: Muse, Biopot, Mock.
* Supporting auto connect and background auto connect with SpeziBluetooth backed devices (in our case the Biopot)
* Added initial support for visionOS 🎉 (_Note: visionOS builds currently don't include support for Questionnaires_).
* Added new "Detailed Headband Fit" view for Muse devices.
* Improved discovery of Muse devices.

## :books: Documentation
--

## :white_check_mark: Testing
Tests were adjusted and updated.

### Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Supereg** [comment]
@PSchmiedmayer seems like disabling the Password Auto Fill doesn't work (here/right now)? I did the same changes as in https://github.com/StanfordBDHG/.github/pull/47 (e.g., enabling in the action, minor fast lane changes).
Seems like current PRs in the template app have a similar problem judging by their CI output (e.g., https://github.com/StanfordSpezi/SpeziTemplateApplication/pull/68)?

**Comment by PSchmiedmayer** [comment]
@Supereg Wanted to follow up on the password autofill issue; could we isolate this to a dedicated runner that causes this problem?

**Comment by Supereg** [comment]
Didn't have the time to further investigate this issue right now.

**Comment by PSchmiedmayer** [comment]
No worries; I will see what I can do this week 🚀

**Comment by Supereg** [comment]
Okay, the run just went through 🙃

File: NAMS.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 1]
```diff
@@ -122,116 +122,107 @@
...
     },
     {
       "identity" : "spezionboarding",
       "kind" : "remoteSourceControl",
       "location" : "https://github.com/StanfordSpezi/SpeziOnboarding.git",
       "state" : {
-        "revision" : "70d1a740a0da8a5bc83d95bcff177ebd3e7402c0",
-        "version" : "0.7.0"
+        "branch" : "feature/vision-os",
+        "revision" : "a11d24e8917c11c1bc226eb970a7419047a035f7"
```
Comment (Supereg) [inline NAMS.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:198]: Upgrade to SpeziOnboarding release

File: NAMS.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 2]
```diff
@@ -122,116 +122,107 @@
...
     },
     {
       "identity" : "spezibluetooth",
       "kind" : "remoteSourceControl",
       "location" : "https://github.com/StanfordSpezi/SpeziBluetooth",
       "state" : {
-        "revision" : "ae2d554434431ea5be5fc85a8dba8d55aa54c159",
-        "version" : "0.2.0"
+        "branch" : "feature/unit-testing-setup",
+        "revision" : "16ce491c3f24c297ff55351c04bd87402af1a7bf"
```
Comment (Supereg) [inline NAMS.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:162]: Upgrade to release version.

**Pull request merged**